template-processors/base/bin/processParameters.sh

```diff
@@ -17,32 +17,50 @@
 set -o nounset
 set -o errexit
 
-echo Processing Parameters
+echo "Processing Parameters"
 
-# Determine how many yaml files we have
-export YAML_COUNT="$(ls -1 $CLONED_PARAMETER_GIT_DIR/*.{yaml,yml,json} 2> /dev/null | wc -l)"
+FOLDERS=""
 
-# Get the list of yaml files to process
-export YAML_FILES="$(ls $CLONED_PARAMETER_GIT_DIR/*.{yaml,yml,json} 2> /dev/null)"
+if [ -e ${CLONED_PARAMETER_GIT_DIR}/hiera.lst ] ; then
+  echo "Generating hierarchy"
+  # a hierarchy list was provided, so lets process it
+  while read -r DIR
+    do
+      # remove everything after # to allow comments and nuke all whitespaces
+      DIR="$(echo -e "${DIR%%#*}" | tr -d '[:space:]')"
 
-# do a merge if there's more than one yaml file
-if [ "${YAML_COUNT}" -gt 1 ]; then
-  echo "Merging all available yaml files"
-  goyq merge ${YAML_FILES} > $CLONED_PARAMETER_GIT_DIR/eunomia_values_processed1.yaml
-fi
+      if [ -n "${DIR}" ]; then
+        # add the folder to the end of the list
+        FOLDERS="${FOLDERS} ${CLONED_PARAMETER_GIT_DIR}/${DIR}"
+      fi
+    done
+fi <<< $(envsubst < ${CLONED_PARAMETER_GIT_DIR}/hiera.lst)
 
-# if there's just one, make sure it has the proper name
-if [ "${YAML_COUNT}" -eq 1 ]; then
-    mv ${YAML_FILES} $CLONED_PARAMETER_GIT_DIR/eunomia_values_processed1.yaml
-fi
+# Add the current folder as the last one to process
+FOLDERS="${FOLDERS} ${CLONED_PARAMETER_GIT_DIR}"
+
+echo "Going to process parameters file in the following folder(s): ${FOLDERS}"
+
+VALUES_FILE=${CLONED_PARAMETER_GIT_DIR}/eunomia_values_processed1.yaml
+
+# ensure our base file exists with one document to make yq happy
+echo "---" > ${VALUES_FILE}
+
+for DIR in ${FOLDERS}; do
+  echo "Processing files in ${DIR}"
+  
+  # get the list of yaml files to process
+  YAML_FILES="$(find ${DIR} -name \*.json -o -name \*.yaml -o -name \*.yml  -maxdepth 1)"
+
+  # merge the files
+  goyq merge --inplace ${VALUES_FILE} ${YAML_FILES}
+done
```

Comment:
[thinking] Maybe we could consider merging the above blocks into a single pipeline more or less like below:

```bash
echo "---" > "$VALUES_FILE"
cat "$CLONED_PARAMETER_GIT_DIR/hiera.lst" |
    sed -e 's/[[:space:]]*#.*//' |  # delete comments
    grep '\S' |   # prune empty lines
    envsubst |
    (echo "Processing parameters from the following folder(s): " >&2; tee /dev/stderr) |  # print debugging info on stderr
    xargs -n1 -I@ find "$CLONED_PARAMETER_GIT_DIR/@" -name \*.json -o -name \*.yaml -o -name \*.yml -maxdepth 1 |
    xargs goyq merge --inplace "$VALUES_FILE"
```

Though OTOH, it has a notable disadvantage that the debugging info we'd be getting with this change seems of lower quality :/

Reply:
That's how I had it originally, but decided to use the current approach to make debugging easier.